src/hotspot/cpu/aarch64/templateTable_aarch64.cpp

```diff
@@ -2357,32 +2366,107 @@ void TemplateTable::load_resolved_field_entry(Register obj,
   }
 }
 
-// The Rcache and index registers must be set before call
-// n.b unlike x86 cache already includes the index offset
-void TemplateTable::load_field_cp_cache_entry(Register obj,
-                                              Register cache,
-                                              Register index,
-                                              Register off,
-                                              Register flags,
-                                              bool is_static = false) {
-  assert_different_registers(cache, index, flags, off);
+void TemplateTable::load_resolved_method_entry_special_or_static(Register cache,
+                                                                 Register method,
+                                                                 Register flags) {
 
-  ByteSize cp_base_offset = ConstantPoolCache::base_offset();
-  // Field offset
-  __ ldr(off, Address(cache, in_bytes(cp_base_offset +
-                                          ConstantPoolCacheEntry::f2_offset())));
-  // Flags
-  __ ldrw(flags, Address(cache, in_bytes(cp_base_offset +
-                                           ConstantPoolCacheEntry::flags_offset())));
+  // setup registers
+  const Register index = r4;
```

Comment:
This exists in x86 as well in each of the `load_resolved_method_entry_...()` methods. Some of these only have three arguments which cannot be reused so there is the option to include `index` as an argument, but this introduces an inconsistency among these similar methods. 

Should all of these methods take `index` which can be a reused register?

Reply:
If you add an extra argument then please name it `temp` or something equally clear as to the fact that this is for storing a local scratch register rather than an input/output. n.b. this is the main reason for resisting the urge to move register declarations+initializations up the call chain. It makes it less obvious that a register is only being used local to a callee.

Reply:
I think `index` is only used before `flags` and `method` get assigned. Can't we use one of them?
Otherwise, passing `temp` would be fine, too.